daemon/compartment.c

```diff
@@ -1176,11 +1201,11 @@ compartment_start(compartment_t *compartment)
 {
 	ASSERT(compartment);
 
-	if ((compartment_get_state(compartment) != COMPARTMENT_STATE_STOPPED) &&
-	    (compartment_get_state(compartment) != COMPARTMENT_STATE_REBOOTING)) {
+	// check if container is startable
+	if (!compartment_is_startable(compartment)) {
 		ERROR("Container %s is not stopped and can therefore not be started",
 		      compartment_get_description(compartment));
-		return COMPARTMENT_ERROR;
+		return -1;
```

Comment:
do not change "return COMPARTMENT_ERROR". In other error cases compartment_start returns COMPARTMENT_ERROR, too

Reply:
remove check completly. since container_is_startable() should be called from outside before calling the internal compartment_start function